### PR TITLE
MSFS native support

### PR DESCRIPTION
**Pull request by tzobler**
All necessary fields in the GUI and mapping tool have been created (based on Roland's changes). Tested communication with datastore and no errors so far. "Saved settings" message added when saving distributor selection. Search for aircaft.cfg optimized for MSFS

### CODE REVIEW

File: src/blackmisc/simulation/fscommon/fscommonutil.cpp [hunk 1]
```diff
@@ -182,6 +182,14 @@ namespace BlackMisc::Simulation::FsCommon
         return CFsCommonUtil::validateSimObjectsPath(QSet<QString>(simObjectPaths.begin(), simObjectPaths.end()), models, validModels, invalidModels, ignoreEmptyFileNames, stopAtFailedFiles, stopped);
     }
 
+    CStatusMessageList CFsCommonUtil::validateMSFSSimObjectsPath(const CAircraftModelList &models, CAircraftModelList &validModels, CAircraftModelList &invalidModels, bool ignoreEmptyFileNames, int stopAtFailedFiles, std::atomic_bool &stopped, const QString &simulatorDir)
+    {
+        Q_UNUSED(simulatorDir)
+        // TODO TZ
```
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fscommonutil.cpp:188]: Is this TODO still outstanding?

File: src/blackmisc/simulation/fscommon/fsdirectories.cpp [hunk 2]
```diff
@@ -88,6 +89,59 @@ namespace BlackMisc::Simulation::FsCommon
         return dir;
     }
 
+    QString msfsDirImpl()
+    {
+        QStringList someDefaultDirs(
+            {
+                // C:\Users\rolan\AppData\Local\Packages\Microsoft.FlightSimulator_8wekyb3d8bbwe\
+            // todo msfs: add default simulator directory
+            });
```
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fsdirectories.cpp:98]: This variable is unused.

File: src/blackmisc/simulation/fscommon/fsdirectories.cpp [hunk 3]
```diff
@@ -88,6 +89,59 @@ namespace BlackMisc::Simulation::FsCommon
...
+        const QStringList locations = QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation);
+        for (const QString &path : locations)
+        {
+            const QString msfsPackage = CFileUtils::appendFilePaths(CFileUtils::appendFilePaths(path, "Packages"), "Microsoft.FlightSimulator_8wekyb3d8bbwe");
+            const QDir d(msfsPackage);
+            if (!d.exists()) { continue; }
+            return msfsPackage;
+        }
```
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fsdirectories.cpp:106]: This likely only works for some installation methods of MSFS (according to https://helpdesk.aerosoft.com/hc/en-gb/articles/5023507568925-How-to-locate-the-Community-folder-in-Microsoft-Flight-Simulator only for Microsoft Store installations).

Other installation directories might be harder to locate automatically (perhaps through some registry keys). If this method is merge like this into ``main``, we should open an issue to improve it in the future.

File: src/blackmisc/simulation/fscommon/fsdirectories.cpp [hunk 4]
```diff
@@ -108,18 +162,39 @@ namespace BlackMisc::Simulation::FsCommon
         return CFsDirectories::fsxSimObjectsDirFromSimDir(dir);
     }
 
+    // TODO TZ
```
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fsdirectories.cpp:165]: Is this TODO still outstanding?

File: src/blackmisc/simulation/fscommon/fsdirectories.cpp [hunk 5]
```diff
@@ -108,18 +162,39 @@ namespace BlackMisc::Simulation::FsCommon
...
+        return dir;
+    }
+
     QString CFsDirectories::fsxSimObjectsDirFromSimDir(const QString &simDir)
     {
         if (simDir.isEmpty()) { return {}; }
         return CFileUtils::appendFilePaths(CFileUtils::normalizeFilePathToQtStandard(simDir), "SimObjects");
     }
 
+    // TODO TZ
```
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fsdirectories.cpp:191]: Is this TODO still outstanding?

File: src/blackmisc/simulation/fscommon/fsdirectories.cpp [hunk 6]
```diff
@@ -507,6 +604,25 @@ namespace BlackMisc::Simulation::FsCommon
         return files;
     }
 
+    // TODO TZ
```
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fsdirectories.cpp:607]: Is this TODO still outstanding?

File: src/blackmisc/simulation/fscommon/fsdirectories.cpp [hunk 7]
```diff
@@ -517,6 +633,17 @@ namespace BlackMisc::Simulation::FsCommon
         return paths;
     }
 
+    // TODO TZ
```
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fsdirectories.cpp:636]: Is this TODO still outstanding?

File: src/blackmisc/simulation/fscommon/fsdirectories.cpp [hunk 8]
```diff
@@ -565,6 +692,55 @@ namespace BlackMisc::Simulation::FsCommon
...
```
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fsdirectories.cpp:695]: Is this TODO still outstanding?

File: src/blackmisc/simulation/fscommon/fsdirectories.cpp
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fsdirectories.cpp:1]: Understanding the code to find the MSFS paths is a bit complicated. Maybe more comments can be added. For example some example paths like already done in the header, expected input and output path of the method, ... Or perhaps even add some unittests.

File: src/blackmisc/simulation/fscommon/fsdirectories.h [hunk 9]
```diff
@@ -37,15 +37,25 @@ namespace BlackMisc::Simulation::FsCommon
         //! FSX's simobject dir, resolved from multiple sources
         static const QString &fsxSimObjectsDir();
 
+        // TODO TZ
```
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fsdirectories.h:40]: Is this TODO still outstanding?

File: src/blackmisc/simulation/fscommon/fsdirectories.h [hunk 10]
```diff
@@ -110,18 +126,37 @@ namespace BlackMisc::Simulation::FsCommon
         //! Get all the SimObjects paths from all config files
         static QSet<QString> allFsxSimObjectPaths();
 
+        // TODO TZ
```
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fsdirectories.h:129]: Is this TODO still outstanding?

File: src/blackmisc/simulation/fscommon/fsdirectories.h [hunk 11]
```diff
@@ -110,18 +126,37 @@ namespace BlackMisc::Simulation::FsCommon
...
+        //! Get all the SimObjects paths from all config files
+        static QSet<QString> allMsfsSimObjectPaths();
+
         //! Find the config files (fsx.cfg)
         // C:/Users/Joe Doe/AppData/Roaming/Microsoft/FSX/fsx.cfg
         static QStringList findFsxConfigFiles();
 
+        // TODO TZ
```
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fsdirectories.h:137]: Is this TODO still outstanding?

File: src/blackmisc/simulation/fscommon/fsdirectories.h [hunk 12]
```diff
@@ -110,18 +126,37 @@ namespace BlackMisc::Simulation::FsCommon
...
+        //! Find the config files (fsx.cfg)
+        // C:/Users/Joe Doe/AppData/Roaming/Microsoft/FSX/fsx.cfg
+        static QStringList findMsfsConfigFiles();
+
         //! Get all the SimObjects paths from fsx.cfg
         // SimObjectPaths.0=SimObjects\Airplanes
         static QSet<QString> fsxSimObjectsPaths(const QStringList &fsxFiles, bool checked);
 
+        // TODO TZ
```
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fsdirectories.h:146]: Is this TODO still outstanding?

File: src/blackmisc/simulation/fscommon/fsdirectories.h [hunk 13]
```diff
@@ -110,18 +126,37 @@ namespace BlackMisc::Simulation::FsCommon
...
+        //! Get all the SimObjects paths from msfs.cfg
+        // SimObjectPaths.0=SimObjects\Airplanes
+        static QSet<QString> msfsSimObjectsPaths(const QStringList &msfsFiles, bool checked);
+
         //! Get all the SimObjects files from fsx.cfg
         // SimObjectPaths.0=SimObjects\Airplanes
         static QSet<QString> fsxSimObjectsPaths(const QString &fsxFile, bool checked);
 
+        // TODO TZ
```
Comment (ltoenning) [inline src/blackmisc/simulation/fscommon/fsdirectories.h:155]: Is this TODO still outstanding?

File: src/blackmisc/simulation/aircraftmodelutils.cpp [hunk 14]
```diff
@@ -179,6 +179,12 @@ namespace BlackMisc::Simulation
                 const CStatusMessageList specificTests2 = FsCommon::CFsCommonUtil::validateFSXSimObjectsPath(models, validModels, invalidModels, ignoreEmpty, stopAtFailedFiles, wasStopped, simulatorDir);
                 specificTests.push_back(specificTests2);
             }
+            else if (simulator.isMSFS())
+            {
+                // todo msfs: MSFS specific model validation
+                // const CStatusMessageList specificTests2 = FsCommon::CFsCommonUtil::validateMSFSSimObjectsPath(models, validModels, invalidModels, ignoreEmpty, stopAtFailedFiles, wasStopped, simulatorDir);
+                // specificTests.push_back(specificTests2);
```
Comment (ltoenning) [inline src/blackmisc/simulation/aircraftmodelutils.cpp:186]: Open a GitHub issue for this, so it does not get lost.

File: src/blackgui/components/dblogincomponent.ui [hunk 15]
```diff
@@ -382,8 +382,8 @@ p, li { white-space: pre-wrap; }
...
-&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;If you want to use VATSIM SSO, you need to enable it once ##swiftEnableSSO## &lt;/li&gt;
-&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;In case you do not have a password you can try &amp;quot;anonymous&amp;quot;/&amp;quot;anonymous&amp;quot;. No guarantee it will work, but worth a trial. &lt;/li&gt;&lt;/ol&gt;&lt;/body&gt;&lt;/html&gt;</string>
+&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;VATSIM SSO is not possible anymore&lt;/li&gt;
+&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;In case you do not have a password you can request access data in the data store </string>
```
Comment (ltoenning) [inline src/blackgui/components/dblogincomponent.ui:386]:
```suggestion
&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;If you do not have credentials, you can request access via the data store.</string>
```

**Review by ltoenning — CHANGES_REQUESTED** [review]

File: src/blackcore/fsd/fsdclient.cpp [hunk 16]
```diff
@@ -124,6 +124,7 @@ namespace BlackCore::Fsd
 
         if (!m_statistics && (CBuildConfig::isLocalDeveloperDebugBuild() || (sApp && sApp->getOwnDistribution().isRestricted())))
         {
+            // TOTO TZ
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackcore/fsd/fsdclient.cpp:127]: Comment will be deleted

File: src/blackconfig/buildconfig.h [hunk 17]
```diff
@@ -38,6 +38,9 @@ namespace BlackConfig
         //! with FG support?
         static constexpr bool isCompiledWithFGSupport(); // defined in buildconfig_gen.inc.in
 
+        //! with MSFS support?
+        static constexpr bool isCompiledWithMSFSSupport(); // defined in buildconfig_gen.inc.in
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackconfig/buildconfig.h:42]: I suggest MSFS20 and later MSFS24. Agreed?

File: resources/share/shared/bootstrap/bootstrap.json
[the remark this answers is not in the document]
Comment (tzobler) [reply resources/share/shared/bootstrap/bootstrap.json:1]: I only made temporary changes to the file for testing with the new database. Original file can be retained

File: src/blackmisc/simulation/aircraftmodelloaderprovider.h [hunk 18]
```diff
@@ -1,5 +1,10 @@
-// SPDX-FileCopyrightText: Copyright (C) 2015 swift Project Community / Contributors
-// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-swift-pilot-client-1
+/* Copyright (C) 2015
+ * swift project Community / Contributors
+ *
+ * This file is part of swift project. It is subject to the license terms in the LICENSE file found in the top-level
+ * directory of this distribution. No part of swift project, including this file, may be copied, modified, propagated,
+ * or distributed except according to the terms contained in the LICENSE file.
+ */
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/simulation/aircraftmodelloaderprovider.h]: completed

File: src/blackmisc/simulation/data/modelcaches.cpp [hunk 19]
```diff
@@ -1,5 +1,10 @@
-// SPDX-FileCopyrightText: Copyright (C) 2016 swift Project Community / Contributors
-// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-swift-pilot-client-1
+/* Copyright (C) 2016
+ * swift project community / contributors
+ *
+ * This file is part of swift project. It is subject to the license terms in the LICENSE file found in the top-level
+ * directory of this distribution. No part of swift project, including this file, may be copied, modified, propagated,
+ * or distributed except according to the terms contained in the LICENSE file.
+ */
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/simulation/data/modelcaches.cpp]: completed

File: src/blackmisc/simulation/data/modelcaches.h [hunk 20]
```diff
@@ -1,5 +1,10 @@
-// SPDX-FileCopyrightText: Copyright (C) 2016 swift Project Community / Contributors
-// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-swift-pilot-client-1
+/* Copyright (C) 2016
+ * swift project community / contributors
+ *
+ * This file is part of swift project. It is subject to the license terms in the LICENSE file found in the top-level
+ * directory of this distribution. No part of swift project, including this file, may be copied, modified, propagated,
+ * or distributed except according to the terms contained in the LICENSE file.
+ */
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/simulation/data/modelcaches.h]: completed

File: src/blackmisc/simulation/distributor.cpp [hunk 21]
```diff
@@ -1,5 +1,10 @@
-// SPDX-FileCopyrightText: Copyright (C) 2015 swift Project Community / Contributors
-// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-swift-pilot-client-1
+/* Copyright (C) 2015
+ * swift project Community / Contributors
+ *
+ * This file is part of swift project. It is subject to the license terms in the LICENSE file found in the top-level
+ * directory of this distribution. No part of swift project, including this file, may be copied, modified, propagated,
+ * or distributed except according to the terms contained in the LICENSE file.
+ */
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/simulation/distributor.cpp]: completed

File: src/blackmisc/simulation/distributorlistpreferences.h [hunk 22]
```diff
@@ -1,5 +1,10 @@
-// SPDX-FileCopyrightText: Copyright (C) 2016 swift Project Community / Contributors
-// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-swift-pilot-client-1
+/* Copyright (C) 2016
+ * swift project Community / Contributors
+ *
+ * This file is part of swift project. It is subject to the license terms in the LICENSE file found in the top-level
+ * directory of this distribution. No part of swift project, including this file, may be copied, modified, propagated,
+ * or distributed except according to the terms contained in the LICENSE file.
+ */
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/simulation/distributorlistpreferences.h]: completed

File: src/blackmisc/simulation/fscommon/fscommonutil.cpp (hunk 1 above)
Comment (tzobler) [reply src/blackmisc/simulation/fscommon/fscommonutil.cpp:188]: completed

File: src/blackmisc/simulation/settings/simulatorsettings.h [hunk 23]
```diff
@@ -1,5 +1,10 @@
-﻿// SPDX-FileCopyrightText: Copyright (C) 2016 swift Project Community / Contributors
-// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-swift-pilot-client-1
+﻿/* Copyright (C) 2016
+ * swift project community / contributors
+ *
+ * This file is part of swift project. It is subject to the license terms in the LICENSE file found in the top-level
+ * directory of this distribution. No part of swift project, including this file, may be copied, modified, propagated,
+ * or distributed except according to the terms contained in the LICENSE file.
+ */
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/simulation/settings/simulatorsettings.h]: completed

File: src/blackmisc/simulation/simulatorinfo.h [hunk 24]
```diff
@@ -198,8 +198,8 @@ namespace BlackMisc::Simulation
         //! From database JSON
         static CSimulatorInfo fromDatabaseJson(const QJsonObject &json, const QString &prefix);
 
-        //! @{
         //! Const simulator info objects
+        //! @{
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/simulation/simulatorinfo.h:202]: I didn't consciously change the order. MSFS now to XPLANE. Does that fit then?

File: src/blackmisc/simulation/simulatorplugininfo.cpp [hunk 25]
```diff
@@ -1,4 +1,4 @@
-// SPDX-FileCopyrightText: Copyright (C) 2013 swift Project Community / Contributors
+// SPDX-FileCopyrightText: Copyright (C) 2018 swift Project Community / Contributors
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/simulation/simulatorplugininfo.cpp:1]: completed

File: src/plugins/simulator/fsxcommon/simconnectdatadefinition.cpp [hunk 26]
```diff
@@ -289,6 +294,18 @@ namespace BlackSimPlugin::FsxCommon
...
+    {
+        HRESULT hr = s_ok();
+        hr += SimConnect_AddToDataDefinition(hSimConnect, CSimConnectDefinitions::DataTransponderModeMSFS, "TRANSPONDER STATE:1", "Enum");
+        hr += SimConnect_AddToDataDefinition(hSimConnect, CSimConnectDefinitions::DataTransponderModeMSFS, "TRANSPONDER IDENT:1", "Bool");
+        if (isFailure(hr))
+        {
+            CLogMessage(static_cast<CSimConnectDefinitions *>(nullptr)).error(u"SimConnect error: MSFS transponder data definitions %1") << hr;
+        }
+        return hr;
+    }
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/plugins/simulator/fsxcommon/simconnectdatadefinition.cpp:307]: Yes, the transponder switch has been confirmed and users have commented very positively on it

File: src/plugins/simulator/fsxcommon/simconnectdatadefinition.h [hunk 27]
```diff
@@ -1,4 +1,4 @@
-// SPDX-FileCopyrightText: Copyright (C) 2013 swift Project Community / Contributors
+// SPDX-FileCopyrightText: Copyright (C) 2018 swift Project Community / Contributors
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/plugins/simulator/fsxcommon/simconnectdatadefinition.h:1]: completed

File: src/blackmisc/simulation/simulatorinfo.h (hunk 24 above)
[the remark this answers is not in the document]
Comment (ltoenning) [reply src/blackmisc/simulation/simulatorinfo.h:202]: This isn't about the order of the simulators. Just if the comment ``//! @{`` should be **before** or **after** the actual text comment. Previously ``//! @{`` was put before the text comment. Now it is put after the text comment.
According to #188, this has a different meaning (group comment vs. same comment for all members inside the braces).

File: src/plugins/simulator/fsxcommon/simconnectdatadefinition.h [hunk 28]
```diff
@@ -105,8 +106,8 @@ namespace BlackSimPlugin::FsxCommon
         char atcAirline[64]; //!< Airline used by ATC
         char atcFlightNumber[8]; //!< Flight Number used by ATC
 
-        //! @{
         //! Copy the strings, length from docu
+        //! @{
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/plugins/simulator/fsxcommon/simconnectdatadefinition.h:110]: this change is not new. I don't recognize the problem

File: src/plugins/simulator/fsxcommon/simulatorfsxsimconnectproc.cpp [hunk 29]
```diff
@@ -1,4 +1,4 @@
-// SPDX-FileCopyrightText: Copyright (C) 2013 swift Project Community / Contributors
+// SPDX-FileCopyrightText: Copyright (C) 2018 swift Project Community / Contributors
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/plugins/simulator/fsxcommon/simulatorfsxsimconnectproc.cpp:1]: done

File: src/plugins/simulator/fsxcommon/simconnectdatadefinition.h (hunk 28 above)
[the remark this answers is not in the document]
Comment (ltoenning) [reply src/plugins/simulator/fsxcommon/simconnectdatadefinition.h:110]: This is likely a rebase error, as well as all other changes where the comments were swapped.
The order was changed on the main branch with https://github.com/swift-project/pilotclient/commit/7691e42a5b1a526372505538c5c3afb9d77789ea some time ago. This wasn't included in the "old" MSFS branch.

The changes to the comments on the "new" MSFS should be reverted.

File: src/blackconfig/buildconfig.h [hunk 30]
```diff
@@ -38,6 +38,9 @@ namespace BlackConfig
...
+
         //! with FSUIPC support?
         static constexpr bool isCompiledWithFsuipcSupport(); // defined in buildconfig_gen.inc.in
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackconfig/buildconfig.h:46]: yes, I'll upload it with the next PUSH

File: src/blackmisc/blackmisc.qrc [hunk 31]
```diff
@@ -582,6 +582,7 @@
     <qresource prefix="/simulators">
         <file>icons/simulators/FS9.png</file>
         <file>icons/simulators/FSX.png</file>
+		<file>icons/simulators/MSFS.png</file>
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/blackmisc.qrc:585]: done

File: src/blackmisc/icon.cpp [hunk 32]
```diff
@@ -1,4 +1,4 @@
-// SPDX-FileCopyrightText: Copyright (C) 2013 swift Project Community / Contributors
+// SPDX-FileCopyrightText: Copyright (C) 2018 swift Project Community / Contributors
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/icon.cpp:1]: done

File: src/blackmisc/icons.h [hunk 33]
```diff
@@ -1,4 +1,4 @@
-// SPDX-FileCopyrightText: Copyright (C) 2013 swift Project Community / Contributors
+// SPDX-FileCopyrightText: Copyright (C) 2018 swift Project Community / Contributors
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/icons.h:1]: done

File: src/blackmisc/icons/simulators/MSFS.png
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/icons/simulators/MSFS.png:1]: What do you mean by that ?

File: src/blackmisc/icons.h [hunk 34]
```diff
@@ -503,32 +505,38 @@ namespace BlackMisc
         // -- Simulators
         // --------------------------------------------------
 
-        //! @{
         //! Simulator FSX
+        //! @{
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/icons.h:509]: modified sequence

File: src/blackmisc/simulation/aircraftmodellist.h [hunk 35]
```diff
@@ -295,10 +295,10 @@ namespace BlackMisc
             //! Which simulators are supported in this model list
             CSimulatorInfo simulatorsSupported() const;
 
-            //! Is this here a FS family (P3D/FSX/FS9) model list?
+            //! Is this here a FS family (P3D/FSX/FS9/MSFS) model list?
             bool isLikelyFsFamilyModelList() const;
 
-            //! Is this here a FS family (P3D/FSX) model list?
+            //! Is this here a FS family (P3D/FSX/MSFS) model list?
             bool isLikelyFsxFamilyModelList() const;
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/simulation/aircraftmodellist.h:302]: done. Please check again

File: src/blackmisc/simulation/aircraftmodelloaderprovider.h [hunk 36]
```diff
@@ -27,13 +32,14 @@ namespace BlackMisc::Simulation
         //! Singleton
         static CMultiAircraftModelLoaderProvider &multiModelLoaderInstance();
 
-        //! @{
         //! Simulator specific loaders
+        //! @{
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/simulation/aircraftmodelloaderprovider.h:31]: done. Please check again

File: src/blackmisc/simulation/data/modelcaches.h [hunk 37]
```diff
@@ -290,13 +310,14 @@ namespace BlackMisc::Simulation::Data
         //! Emit cacheChanged() utility function (allows breakpoint)
         void emitCacheChanged(const CSimulatorInfo &simulator);
 
-        //! @{
         //! Cache synchronized flag
+        //! @{
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/simulation/data/modelcaches.h:309]: done, please check again

File: src/blackmisc/simulation/data/modelcaches.h [hunk 38]
```diff
@@ -389,32 +412,14 @@ namespace BlackMisc::Simulation::Data
     public:
         //! \name Interface implementations
         //! @{
-
-        //! \copydoc BlackMisc::Simulation::Data::IMultiSimulatorModelCaches::getCachedModels
```
[the remark this answers is not in the document]
Comment (tzobler) [reply src/blackmisc/simulation/data/modelcaches.h:393]: Done, please check again

File: src/blackmisc/simulation/fscommon/fsdirectories.h (hunk 11 above)
Comment (tzobler) [reply src/blackmisc/simulation/fscommon/fsdirectories.h:140]: I think this function doesn't do what it's supposed to do. That's why I want to watch it

Comment (tzobler) [reply src/blackmisc/simulation/fscommon/fsdirectories.h:140]: I think this function doesn't do what it's supposed to do. That's why I want to watch it

File: src/blackmisc/simulation/fscommon/fsdirectories.h (hunk 12 above)
Comment (tzobler) [reply src/blackmisc/simulation/fscommon/fsdirectories.h:149]: I think this function doesn't do what it's supposed to do. That's why I want to watch it

File: src/blackmisc/simulation/fscommon/fsdirectories.h (hunk 13 above)
Comment (tzobler) [reply src/blackmisc/simulation/fscommon/fsdirectories.h:158]: I think this function doesn't do what it's supposed to do. That's why I want to watch it

**Pull request merged**